Text/CustomEntitySearch/CustomEntitySearch.cs

```diff
@@ -52,48 +53,273 @@ public static class CustomEntitySearch
+                                currMismatch += 1;
+                                wordFound.Write(text.ElementAt(currTextChar));
+                                currTextChar++;
+                                currWordChar++;
                             }
 
+                            if((currWordChar >= wordCharArray.Count || (currTextChar >= textCharArray.Count
+                                && currMismatch + (wordCharArray.Count - currWordChar) <= leniency))  && currMismatch <= leniency)
+                            {
+                                if (!entityMatch.Where(x => (x.Index >= currTextChar - (wordCharArray.Count + leniency)) &&
+                                    (x.Index <= currTextChar - wordCharArray.Count)).Any())
+                                {
+                                    addWord = true;
+                                    valuesList.Add(wordFound.ToString());
+                                    offsetList.Add(currTextChar - wordFound.ToString().Length);
+                                    confidenceList.Add(currMismatch + (wordCharArray.Count - currWordChar) / leniency);
+                                    if (minVal < 0 || wordFound.ToString().Length < minVal)
+                                        minVal = wordFound.ToString().Length;
+                                }
+                                currWordChar = KMPTable[currWordChar];
+                                wordFound.GetStringBuilder().Clear();
+                                currMismatch = 0;
+                            }
                         }
                     }
+                    else
+                    {
+                        currWordChar = KMPTable[currWordChar];
+                        if (currWordChar < 0)
+                        {
+                            currWordChar++;
+                            currTextChar++;
+                        }
+                        wordFound.GetStringBuilder().Clear();
+                        currMismatch = 0;
+                    }
+                }
 
-                    outRecord.Data["Entities"] = entities;
-                    outRecord.Data["EntitiesFound"] = entitiesFound;
-                    return outRecord;
-                });
-
-            return new OkObjectResult(response);
+                for (int i = 0; i < valuesList.Count; i++)
+                {
+                    if (valuesList[i].Length == minVal)
+                    {
+                        entities.Add(
+                            new Entity
+                            {
+                                Category = "customEntity",
+                                Value = valuesList[i],
+                                Offset = offsetList[i],
+                                Confidence = 1 - confidenceList[i]
+                            });
+                    }
+                }
+            }
+            if (addWord)
+                entitiesFound.Add(checkMatch);
```

Comment:
why do you have this boolean? can you just call entitiesFound.Add(checkMatch); whenever you call entities.Add(~)?

Reply:
No since depending on fuzzy/exact/multiple entities in the text, i only add entitiesFound once